ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh

```diff
@@ -1,21 +1,126 @@
 #!/usr/bin/env bash
 
 set -x
+set +eu
+
+BASE_DIR=${HOME:-"/alabama"}
+MUST_GATHER_TIMEOUT=${MUST_GATHER_TIMEOUT:-"5m"}
+NS_OPERATORS=${NS_OPERATORS:-"openstack"}
+NS_SERVICES=${NS_SERVICES:-"openstack"}
+
+# Creates directory if does not exists
+#  Parameters:
+#  $1: directory to be created
+function create_out_dir {
+  local OUT_DIR=${1}
+  if [ ! -d "./${OUT_DIR}" ]; then
+    mkdir -p ${OUT_DIR}
+  fi
+}
+
+# Generates a debug command tp get pods logs and description. To be used with "xargs -I {}".
+#  Parameters:
+#  $1: directory to store output logs (optional)
+function gen_pods_debug_cmd {
+  local cmd
+  if [ -z "${1}" ]; then
+    cmd=("echo 'Logs for pod: {}'; oc describe pod {} > {}-describe.log; oc logs --all-containers=true {} >> {}-containers.log")
+  else
+    create_out_dir ${1}
+    cmd=("echo 'Logs for pod: {}'; oc describe pod {} > ${1}/{}-describe.log; oc logs --all-containers=true {} >> ${1}/{}-containers.log")
+  fi
+  echo "${cmd[@]}"
+}
+
+# Generates a debug command to get yaml output for a resource. To be used with "xargs -I {}".
+#  Parameters:
+#  $1: directory to store output logs (optional)
+function gen_resource_yaml_debug_cmd {
+  local cmd
+  if [ -z "${1}" ]; then
+    cmd=("echo 'Getting info for {} resource'; oc get {} -o yaml > {}.yaml")
+  else
+    create_out_dir ${1}
+    cmd=("echo 'Getting info for {} resource'; oc get {} -o yaml > ${1}/{}.yaml")
+  fi
+  echo "${cmd[@]}"
+}
+
+# Get all yaml outputs from resource in "oc get all" result.
+#  Parameters:
+#  $1: namespace
+#  $2: string to be used in egrep to filter results (optional)
+function get_all_from_ns {
+  local get_all
+  local cmd
+  local dir
+  if [ -z "${2}" ]; then
+    get_all=$(oc get all -o name -n "${1}")
+  else
+    get_all=$(oc get all -o name -n "${1}" | egrep -iv "${2}")
+  fi
+
+  cmd=$(gen_resource_yaml_debug_cmd)
+  for i in ${get_all}; do
+    dir=$(echo $i | cut -d "/" -f1)
+    create_out_dir "${dir}"
+    echo ${i} | xargs -n1 -I {} sh -c "${cmd[@]}"
```

Comment:
Yeah, I'm using it here just to use its replace `-I` . I could use `sed` instead, but it was giving headaches since ${i} contains slash keys. This was simpler